File: src/JsBind.Net/content/JsBindScripts/Modules/References/ObjectReference.js
import ReferenceBase, { IsReferenceBase } from "./ReferenceBase.js";
import ReferenceType from "./ReferenceType.js";

export default class ObjectReference extends ReferenceBase {
  /** @type {string} */
  accessPath;
}

/**
 * Checks if a value is an ObjectReference.
 * @param {any} value
 * @returns {value is ObjectReference}
 */
export function IsObjectReference(value) {
  return IsReferenceBase(value) && value.__referenceType === ReferenceType.Object;
}

File: src/JsBind.Net/Internal/References/ObjectReference.cs
using System.Text.Json.Serialization;

namespace JsBind.Net.Internal.References
{
    /// <summary>
    /// Object refence class to be serialized and passed to JavaScript.
    /// </summary>
    internal class ObjectReference : ReferenceBase
    {
        public override ReferenceType ReferenceType => ReferenceType.Object;

        [JsonPropertyName("accessPath")]
        public string? AccessPath { get; }

        public ObjectReference(string? accessPath)
        {
            AccessPath = accessPath;
        }
    }
}
